yt_dlp/utils/_utils.py

```diff
@@ -442,7 +442,7 @@ def find_or_raise(haystack, needle, exc):
     content_start += whole_start + 1
     with HTMLBreakOnClosingTagParser() as parser:
         parser.feed(html[whole_start:content_start])
-        if not parser.tagstack or parser.tagstack[0] != tag:
+        if not parser.tagstack or parser.tagstack[0] != tag.lower():
```

Comment:
this seems insufficient, look at lines 432-441

Reply:
Made some changes to the tests, let me know if there are any other things left.